include/exadg/incompressible_navier_stokes/time_integration/time_int_bdf_coupled_solver.cpp

```diff
@@ -176,8 +176,8 @@ TimeIntBDFCoupled<dim, Number>::do_timestep_solve()
     solution_np = solution_last_iter;
   }
 
-  // update viscosity model
-  if(this->param.viscosity_is_variable())
+  // explicit viscosity update
+  if(this->param.viscosity_is_variable() and not this->param.viscous_term_is_nonlinear())
```

Comment:
Here, I do not like the name `viscous_term_is_nonlinear()`. For me, the question is not whether the term is nonlinear or linear, but whether the user decides for an implicit vs. explicit treatment of this term, or do I misunderstand something?

To avoid the negation, a statement like `if TreatmentOfVariableViscosity::Explicit` would be easier to read in my opinion.

Reply:
Note that the function `viscous_term_is_nonlinear()` already includes `viscosity_is_variable()` internally, so this is something to be corrected.

Reply:
Got that, thanks!